spot-contracts/contracts/PerpetualTranche.sol

```diff
@@ -470,12 +471,42 @@ contract PerpetualTranche is
     }
 
     /// @inheritdoc IPerpetualTranche
-    /// @dev CAUTION: Only the whitelisted vault can call this function.
+    /// @dev Only the whitelisted vault can call this function.
+    function claimFees(address to) external override onlyVault afterStateUpdate nonReentrant whenNotPaused {
+        IERC20Upgradeable perp_ = IERC20Upgradeable(address(this));
+        uint256 collectedBal = perp_.balanceOf(address(perp_));
+        if (collectedBal > 0) {
+            perp_.safeTransfer(to, collectedBal);
+        }
+    }
+
+    /// @inheritdoc IPerpetualTranche
+    /// @dev Only the whitelisted vault can call this function.
+    function payProtocolFee() external override onlyVault afterStateUpdate nonReentrant whenNotPaused {
+        uint256 protocolSharePerc = feePolicy.protocolSharePerc();
+        if (protocolSharePerc > 0) {
+            _mint(feePolicy.protocolFeeCollector(), protocolSharePerc.mulDiv(totalSupply(), ONE - protocolSharePerc));
+        }
+    }
+
+    /// @inheritdoc IPerpetualTranche
+    /// @dev Only the whitelisted vault can call this function.
     ///      The logic controlling the frequency and magnitude of debasement should be vetted.
+    ///      Pays a protocol fee on every rebalance.
     function rebalanceToVault(
-        uint256 underlyingAmtToTransfer
+        int256 underlyingAmtToTransfer
     ) external override onlyVault afterStateUpdate nonReentrant whenNotPaused {
-        _mint(address(vault), underlyingAmtToTransfer.mulDiv(totalSupply(), _reserveValue()));
+        // When value is flowing out of perp to the vault, we mint the vault perp tokens.
+        if (underlyingAmtToTransfer < 0) {
```

Comment:
This sign seems backwards to me. If we're "rebalancing to the vault" a positive value, wouldn't that mean that we're transferring value out?

We should also probably have a docstring for this param, if there's any confusion at all.

Reply:
hmm, then I think we can just make `underlyingAmtToTransfer` to unsigned. And on perp simply transfer value if it's non-zero.